python/lsst/daf/butler/_butler_repo_index.py

```diff
@@ -222,10 +228,12 @@ def get_repo_uri(cls, label: str, return_label: bool = False) -> ResourcePath:
         if repo_uri is None:
             if return_label:
                 return ResourcePath(label, forceAbsolute=False)
-            # This should not raise since it worked earlier.
-            try:
-                index_uri = str(cls._get_index_uri())
-            except KeyError:
-                index_uri = "<environment variable not defined>"
-            raise KeyError(f"Label '{label}' not known to repository index at {index_uri}")
+            raise KeyError(f"Label '{label}' not known to repository index")
         return ResourcePath(repo_uri)
+
+    @classmethod
+    def _validate_configuration(cls, obj: Any) -> dict[str, str]:
```

Comment:
Can we validate once on first read and then know that we have a validated dict in the cache?